OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/alias/AbstractAliasAnalysis.scala

```diff
@@ -0,0 +1,162 @@
+/* BSD 2-Clause License - see OPAL/LICENSE for details. */
+package org.opalj
+package tac
+package fpcf
+package analyses
+package alias
+
+import scala.annotation.tailrec
+
+import org.opalj.br.ArrayType
+import org.opalj.br.BaseType
+import org.opalj.br.ClassHierarchy
+import org.opalj.br.ObjectType
+import org.opalj.br.ReferenceType
+import org.opalj.br.fpcf.FPCFAnalysis
+import org.opalj.br.fpcf.properties.alias.Alias
+import org.opalj.br.fpcf.properties.alias.AliasEntity
+import org.opalj.br.fpcf.properties.alias.NoAlias
+import org.opalj.fpcf.Entity
+import org.opalj.fpcf.InterimResult
+import org.opalj.fpcf.ProperPropertyComputationResult
+import org.opalj.fpcf.Result
+import org.opalj.fpcf.SomeEPS
+import org.opalj.value.ValueInformation
+
+/**
+ * A base trait for all alias analyses.
+ */
+trait AbstractAliasAnalysis extends FPCFAnalysis {
+
+    protected[this] type AnalysisContext <: AliasAnalysisContext
+    protected[this] type AnalysisState <: AliasAnalysisState
+    protected[this] type Tac = TACode[TACMethodParameter, DUVar[ValueInformation]]
+
+    /**
+     * Determines the alias relation for the given entity.
+     * @param e The entity to determine the aliasing information for.
+     * @return The result of the computation.
+     */
+    def determineAlias(e: Entity): ProperPropertyComputationResult = {
+        e match {
+            case entity: AliasEntity =>
+                val context = createContext(entity)
+
+                if (checkTypeCompatibility(context)) doDetermineAlias(context, createState)
+                else result(NoAlias)(context)
+            case _ => throw new UnknownError("unhandled entity type")
+        }
+    }
+
+    /**
+     * Determines if the types of the given elements are compatible, i.e. it is possible that they refer to the same objects.
+     * Two elements are compatible if they are both reference types and one is a subtype of the other.
+     * In any other case, e.g., an Integer and a String, the elements cannot possibly refer to the same object.
+     *
+     * @param context The alias analysis context to check the types for.
+     * @return True if the types are compatible, false otherwise.
+     */
+    private[this] def checkTypeCompatibility(context: AliasAnalysisContext): Boolean = {
+
+        if (context.element1.isNullValue || context.element2.isNullValue)
+            return true
+
+        if (!context.element1.isReferenceType || !context.element2.isReferenceType)
+            return false
+
+        if (context.element1.referenceType.isArrayType != context.element2.referenceType.isArrayType)
+            return false
+
+        val type1 = getObjectType(context.element1.referenceType)
+        val type2 = getObjectType(context.element2.referenceType)
+        implicit val classHierarchy: ClassHierarchy = project.classHierarchy
+
+        if (type1.isEmpty || type2.isEmpty)
+            return false
+
+        if (type1.get._2 != type2.get._2)
```

Comment:
I think you should just be able to use ClassHierarchy.isSubtypeOf. It handles array types for you just well.